proto/frequenz/dispatch/v1/dispatch.proto

```diff
@@ -75,6 +75,62 @@ message Dispatch {
 
   // The dispatch payload
   google.protobuf.Struct payload = 11;
+
+  // The recurrence rule
+  RecurrenceRule recurrence_rule = 12;
+}
+
+// Information about the schedule for reoccurring dispatches
+//
+// Example:
+// Run at noon every day: RecurrenceRule {
+//   seconds = [0],
+//   minutes = [0],
+//   hours = [12],
+//   day_of_month = [],
+//   day_of_week = [],
+//   month = [],
+//   year = [],
+//   count = 0,
+//   // Do not set the `until` field
+// }
+message RecurrenceRule {
+  // Time in seconds
+  // Allowed values: 0 - 59
+  repeated uint32 seconds = 1;
+
+  // Time in minutes
+  // Allowed values: 0 - 59
+  repeated uint32 minutes = 2;
+
+  // Time in hours
+  // Allowed values: 0 - 23
+  repeated uint32 hours = 3;
+
+  // Day of month
+  // Allowed values: 1 - (28 - 31, depending on the month/leap year)
+  repeated uint32 month_days = 4;
+
+  // Day of week
+  // Allowed values: 1 - 7 (Mon - Sun)
+  repeated uint32 week_days = 5;
+
+  // Month
+  // Allowed values: 1 - 12
+  repeated uint32 months = 6;
+
+  // Year
+  // Allowed values: Current year and onwards
```

Comment:
If i would want something to recur annually I would need to write out all the years then?

Reply:
We are most likely going to go ahead and just use cron expressions instead of this structure here.

Reply:
Makes sense, missed the comments above.

Reply:
> @thomas-nicolai-frequenz didn't seem to like to use an existing format and wanted everything encoded in the protobuf message itself. Maybe this time we can make him reconsider?

In our discussion about whether to adopt iCal or Cron strings, it's crucial that we address specific technical and usability concerns. My primary concerns revolve around:

1. Support for Underlying Standards: How can we ensure that we support all features of the chosen standard? Do we really support all of them? What does ICal support, what Cron doesn't or vice-verca.

2. Verification Process: There's a significant risk if what's sent through the dispatch API isn't supported by the dispatch actor. Late-stage verification could lead to unnoticed failures or delays. A clear interface would mitigate this risk.

3. Database Storage: The argument favoring string storage in databases due to simplicity isn't strong. We have the capability to serialize any number of protobuf properties in JSON, which enhances index-ability and search-ability in an RDBMS.

Let's base our decision on these concrete issues, rather than personal preferences or perceptions of ease-of-use.

Reply:
> There is also https://pypi.org/project/python-crontab/ / https://github.com/dbader/schedule, but ... and wanted everything encoded in the protobuf message itself. Maybe this time we can make him reconsider?

FYI: Just because we are encoding the parameters in Protobuf explicitly doesn't mean we can't use the third party libraries. The only difference is that instead of providing a string we are simply initialising these libraries with the parameters from the Protobuf message.